parser/prism/Translator.cc

```diff
@@ -18,14 +30,73 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
 
             return make_unique<parser::Pair>(parser.translateLocation(loc), std::move(key), std::move(value));
         }
-        case PM_BLOCK_PARAMETER_NODE: {
+        case PM_BLOCK_ARGUMENT_NODE: { // A block forwarded into a method call, e.g. the `&b` in `a.map(&b)`
+            auto blockArg = reinterpret_cast<pm_block_argument_node *>(node);
+            auto loc = &blockArg->base.location;
+
+            auto expr = translate(blockArg->expression);
+
+            return make_unique<parser::BlockPass>(parser.translateLocation(loc), std::move(expr));
+        }
+        case PM_BLOCK_NODE: { // An explicit block passed to a method call, i.e. `{ ... }` or `do ... end
+            unreachable("PM_BLOCK_NODE has special handling in translateCallWithBlock, see its docs for details.");
+        }
+        case PM_BLOCK_PARAMETER_NODE: { // A block parameter declared at the top of a method, e.g. `def m(&block)`
             auto blockParamNode = reinterpret_cast<pm_block_parameter_node *>(node);
             pm_location_t *loc = &blockParamNode->base.location;
 
             std::string_view name = parser.resolveConstant(blockParamNode->name);
 
             return make_unique<parser::Blockarg>(parser.translateLocation(loc), gs.enterNameUTF8(name));
         }
+        case PM_BLOCK_PARAMETERS_NODE: { // The parameters declared at the top of a PM_BLOCK_NODE
+            auto paramsNode = reinterpret_cast<pm_block_parameters_node *>(node);
+            return translate(reinterpret_cast<pm_node *>(paramsNode->parameters));
```

Comment:
Maybe a comment here that it's not handling block-locals at the moment?

e.g., `foo { |bar; baz, qux| }`

Reply:
Will be addressed in #49